### PR TITLE
Import Point Vector for SuperAnnotate

### PR DESCRIPTION
**Pull request by Fl4m3Ph03n1x**
What?
------

 - This handles the SuperAnnotate integration for Vector Points (they are the same as a `keypoint` in darwin)
 - Changed the name `parse_file` to `parse_path` since sometimes we have to parse full datasets. 
 - Removed validators from the function signature of labelbox
 - Overall doc fixes

How can I test this?
----------------

1. Create a folder with 2 files. One of the files must be called `classes.json`. The other can have any name you want.
2. In the file called `classes.json` write the following content:
```
[
    {
        "attribute_groups": [
            {
                "id": 1,
                "name": "Sex",
                "is_multiselect": false,
                "attributes": [
                    {
                        "id": 1,
                        "name": "Male"
                    },
                    {
                        "id": 2,
                        "name": "Female"
                    }
                ],
                "opened": true,
                "hasChanges": false
            },
            {
                "id": 2,
                "name": "Emotion",
                "is_multiselect": false,
                "attributes": [
                    {
                        "id": 3,
                        "name": "Smiling"
                    },
                    {
                        "id": 4,
                        "name": "Sadness"
                    },
                    {
                        "id": 5,
                        "name": "Crying"
                    }
                ],
                "opened": true,
                "hasChanges": false
            },
            {
                "id": 3,
                "name": "Action",
                "is_multiselect": false,
                "attributes": [
                    {
                        "id": 6,
                        "name": "Skateboarding"
                    },
                    {
                        "id": 7,
                        "name": "Skating"
                    },
                    {
                        "id": 8,
                        "name": "Riding"
                    },
                    {
                        "id": 9,
                        "name": "Running"
                    },
                    {
                        "id": 10,
                        "name": "Swimming"
                    }
                ],
                "opened": true,
                "hasChanges": false
            }
        ],
        "color": "#fd7007",
        "id": 2,
        "name": "Pedestrian",
        "opened": true
    },
    {
        "attribute_groups": [
            {
                "id": 4,
                "name": "Type",
                "is_multiselect": true,
                "attributes": [
                    {
                        "id": 11,
                        "name": "Personal"
                    },
                    {
                        "id": 12,
                        "name": "Large"
                    },
                    {
                        "id": 13,
                        "name": "Occluded"
                    },
                    {
                        "id": 14,
                        "name": "Damaged"
                    }
                ],
                "opened": true,
                "hasChanges": false
            }
        ],
        "color": "#00a8ff",
        "id": 3,
        "name": "Vehicle",
        "opened": true
    },
    {
        "attribute_groups": [],
        "color": "#ff0057",
        "id": 4,
        "name": "Face points",
        "opened": true
    },
    {
        "attribute_groups": [],
        "color": "#734fe2",
        "id": 5,
        "name": "Other",
        "opened": true
    },
    {
        "attribute_groups": [],
        "color": "#51f11e",
        "id": 6,
        "name": "Random",
        "opened": true
    }
]
```

3. In the other file, which can have any name you want as long as it is a `.json` file, write the following:

```
{
    "instances": [
        {
            "type": "point",
            "classId": 4,
            "probability": 100,
            "x": 744.7,
            "y": 732.7,
            "groupId": 0,
            "pointLabels": {},
            "locked": false,
            "visible": true,
            "attributes": []
        },
        {
            "type": "point",
            "classId": 4,
            "probability": 100,
            "x": 849.5,
            "y": 730.3,
            "groupId": 0,
            "pointLabels": {},
            "locked": false,
            "visible": true,
            "attributes": []
        },
        {
            "type": "point",
            "classId": 4,
            "probability": 100,
            "x": 954.3,
            "y": 719,
            "groupId": 0,
            "pointLabels": {},
            "locked": false,
            "visible": true,
            "attributes": []
        },
        {
            "type": "point",
            "classId": 4,
            "probability": 100,
            "x": 1075.1,
            "y": 708.9,
            "groupId": 0,
            "pointLabels": {},
            "locked": false,
            "visible": true,
            "attributes": []
        },
        {
            "type": "point",
            "classId": 4,
            "probability": 100,
            "x": 767.3,
            "y": 781,
            "groupId": 0,
            "pointLabels": {},
            "locked": false,
            "visible": true,
            "attributes": []
        },
        {
            "type": "point",
            "classId": 4,
            "probability": 100,
            "x": 797.7,
            "y": 779.8,
            "groupId": 0,
            "pointLabels": {},
            "locked": false,
            "visible": true,
            "attributes": []
        },
        {
            "type": "point",
            "classId": 4,
            "probability": 100,
            "x": 857.2,
            "y": 788.1,
            "groupId": 0,
            "pointLabels": {},
            "locked": false,
            "visible": true,
            "attributes": []
        },
        {
            "type": "point",
            "classId": 4,
            "probability": 100,
            "x": 959.6,
            "y": 779.8,
            "groupId": 0,
            "pointLabels": {},
            "locked": false,
            "visible": true,
            "attributes": []
        },
        {
            "type": "point",
            "classId": 4,
            "probability": 100,
            "x": 982.8,
            "y": 768.5,
            "groupId": 0,
            "pointLabels": {},
            "locked": false,
            "visible": true,
            "attributes": []
        },
        {
            "type": "point",
            "classId": 4,
            "probability": 100,
            "x": 1044.2,
            "y": 761.9,
            "groupId": 0,
            "pointLabels": {},
            "locked": false,
            "visible": true,
            "attributes": []
        },
        {
            "type": "point",
            "classId": 4,
            "probability": 100,
            "x": 879.8,
            "y": 882.8,
            "groupId": 0,
            "pointLabels": {},
            "locked": false,
            "visible": true,
            "attributes": []
        },
        {
            "type": "point",
            "classId": 4,
            "probability": 100,
            "x": 953.1,
            "y": 875.6,
            "groupId": 0,
            "pointLabels": {},
            "locked": false,
            "visible": true,
            "attributes": []
        },
        {
            "type": "point",
            "classId": 4,
            "probability": 100,
            "x": 831,
            "y": 955.4,
            "groupId": 0,
            "pointLabels": {},
            "locked": false,
            "visible": true,
            "attributes": []
        },
        {
            "type": "point",
            "classId": 4,
            "probability": 100,
            "x": 920.9,
            "y": 932.2,
            "groupId": 0,
            "pointLabels": {},
            "locked": false,
            "visible": true,
            "attributes": []
        },
        {
            "type": "point",
            "classId": 4,
            "probability": 100,
            "x": 1017.4,
            "y": 939.4,
            "groupId": 0,
            "pointLabels": {},
            "locked": false,
            "visible": true,
            "attributes": []
        },
        {
            "type": "point",
            "classId": 4,
            "probability": 100,
            "x": 928.1,
            "y": 1002.5,
            "groupId": 0,
            "pointLabels": {},
            "locked": false,
            "visible": true,
            "attributes": []
        },
        {
            "type": "point",
            "classId": 4,
            "probability": 100,
            "x": 934.6,
            "y": 1106.7,
            "groupId": 0,
            "pointLabels": {},
            "locked": false,
            "visible": true,
            "attributes": []
        }
    ],
    "tags": [],
    "metadata": {
        "version": "1.0.0",
        "name": "sonnie-hiles-PUBt7UPbJFY-unsplash.jpg",
        "status": "Completed"
    }
}
```

4. Follow this guide to point darwin to staging: https://www.notion.so/v7labs/darwin-py-dev-setup-0488c3a235054d579712645d3ff8ae19
5. Pick any image from a dataset you own and change the `name` key inside the `metadata` object from the second file accordingly.
6. `darwin dataset import MY_DATASET superannotate MY_FOLDER_WITH_BOTH_FILES

Ta da !

### CODE REVIEW

File: darwin/importer/formats/superannotate.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+from darwin.datatypes import Annotation, AnnotationClass, AnnotationFile, make_keypoint
+from darwin.importer.formats.superannotate_schemas import (
+    classes_export,
+    superannotate_export,
+)
+
+
+def parse_path(path: Path) -> Optional[AnnotationFile]:
+    """
+    Parses SuperAnnotate annotations inside the given file and returns the corresponding darwin
```
Comment (andreaazzini) [inline darwin/importer/formats/superannotate.py:18]: Not important, but
```suggestion
    Parses SuperAnnotate annotations inside the given file and returns the corresponding Darwin JSON
```

File: darwin/importer/formats/superannotate.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    ValidationError
+        If any given JSON file is malformed or if it has an unknown annotation. 
+        To see a list of possible annotation formats go to:
+        https://doc.superannotate.com/docs/vector-json
+    """
+
+    if not _is_annotation(path):
+        return None
+
+    classes_path = Path(join(path.parent, "classes.json"))
```
Comment (andreaazzini) [inline darwin/importer/formats/superannotate.py:77]:
```suggestion
    classes_path = path.parent / "classes.json"
```

File: darwin/importer/formats/superannotate.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
+import json
+from functools import partial
+from os.path import join
```
Comment (andreaazzini) [inline darwin/importer/formats/superannotate.py:3]: you probably don't need this import here (see later suggestions)

File: darwin/importer/formats/labelbox.py [hunk 4]
```diff
@@ -85,7 +87,7 @@ def parse_file(path: Path, validate: Callable[[Any], None]) -> Optional[List[Ann
 
     with path.open() as f:
         data = json.load(f)
-        validate(data)
+        validate(data, labelbox_export)
```
Comment (andreaazzini) [inline darwin/importer/formats/labelbox.py:90]: for consistency's sake
```suggestion
        validate(data, schema=labelbox_export)
```

File: darwin/importer/formats/superannotate.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+        validate(classes, schema=classes_export)
+
+        with path.open() as annotation_file:
+            data = json.load(annotation_file)
+            validate(data, schema=superannotate_export)
+
+            instances: List[Dict[str, Any]] = data.get("instances")
+            metadata: Dict[str, Any] = data.get("metadata")
+
+            return _convert(instances, path, classes, metadata)
```
Comment (andreaazzini) [inline darwin/importer/formats/superannotate.py:92]: I guess we don't need to keep this block at this level of indentation, we can go one level back

**Review by andreaazzini — CHANGES_REQUESTED** [review]
Brilliant work, very well tested and documented.
A few request changes, but should be quick to fix

File: darwin/importer/formats/superannotate.py (hunk 5 above)
Comment (Fl4m3Ph03n1x) [reply darwin/importer/formats/superannotate.py:92]: How can we go back 1 level? Can I open 2 files at the same time with `with` ?

File: darwin/importer/formats/superannotate.py (hunk 2 above)
Comment (Fl4m3Ph03n1x) [reply darwin/importer/formats/superannotate.py:77]: fixed

File: darwin/importer/formats/superannotate.py (hunk 3 above)
Comment (Fl4m3Ph03n1x) [reply darwin/importer/formats/superannotate.py:3]: fixed

File: darwin/importer/formats/superannotate.py [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
```
Comment (andreaazzini) [inline darwin/importer/formats/superannotate.py:91]:
```suggestion
    with classes_path.open() as classes_file:
        classes = json.load(classes_file)
        validate(classes, schema=classes_export)

    with path.open() as annotation_file:
        data = json.load(annotation_file)
        validate(data, schema=superannotate_export)

    instances: List[Dict[str, Any]] = data.get("instances")
    metadata: Dict[str, Any] = data.get("metadata")

    return _convert(instances, path, classes, metadata)
```

**Review by andreaazzini — CHANGES_REQUESTED** [review]

File: darwin/importer/formats/superannotate.py (hunk 6 above)
Comment (Fl4m3Ph03n1x) [reply darwin/importer/formats/superannotate.py:91]: Ahhh, I get it now !

File: darwin/importer/formats/superannotate.py (hunk 5 above)
Comment (Fl4m3Ph03n1x) [reply darwin/importer/formats/superannotate.py]: done!

**Review by andreaazzini — APPROVED** [review]

**Pull request merged**